src/huggingface_hub/hf_api.py

```diff
@@ -131,45 +131,20 @@ def repo_type_and_id_from_hf_id(hf_id: str):
     return repo_type, namespace, repo_id
 
 
-class RepoObj:
+class RepoFile:
     """
-    HuggingFace git-based system, data structure that represents a file
-    belonging to the current user.
-    """
-
-    def __init__(self, **kwargs):
-        for k, v in kwargs.items():
-            setattr(self, k, v)
-
-    def __repr__(self):
-        items = (f"{k}='{v}'" for k, v in self.__dict__.items())
-        return f"{self.__class__.__name__}({', '.join(items)})"
-
-
-class ModelFile:
-    """
-    Data structure that represents a public file inside a model, accessible from
+    Data structure that represents a public file inside a repo, accessible from
     huggingface.co
-    """
 
-    def __init__(self, rfilename: str, **kwargs):
-        self.rfilename = rfilename  # filename relative to the model root
-        for k, v in kwargs.items():
-            setattr(self, k, v)
-
-    def __repr__(self):
-        items = (f"{k}='{v}'" for k, v in self.__dict__.items())
-        return f"{self.__class__.__name__}({', '.join(items)})"
-
-
-class DatasetFile:
-    """
-    Data structure that represents a public file inside a dataset, accessible
-    from huggingface.co
+    Args:
+        rfilename (str):
+            file name, relative to the repo root. This is the only attribute
+            that's guaranteed to be here, but under certain conditions there can
+            certain other stuff.
     """
 
-    def __init__(self, rfilename: str, **kwargs):
-        self.rfilename = rfilename  # filename relative to the dataset root
+    def __init__(self, *, rfilename: str, **kwargs):
+        self.rfilename = rfilename  # filename relative to the repo root
```

Comment:
I think here we shouldn't put the `*` as:
- There is no decorator
- `rfilename` doesn't have a default so isn't a kwarg

Ok for you if I remove it?

Reply:
(Removed in e31072b, let me know if you want to keep it)